File: prime/prime.py
def isPrime(x):
    x = input("Number:")
    first_factor = None
    i = None
    while i < x:
        searcher = x % i
        if searcher == 0 and first_factor != 0:
            break
        else:
            first_factor = i

        if first_factor == (x - 1):
            print('The number is prime.')
        else:
            print('The number is not prime.')

File: prime/prime.js
let x = prompt("Enter a number:");
function isPrime(x) {
    var firstFactor;
    var i;
    for (i = 2; i < x; i++) {
        let searcher = x % i;
        if ((searcher === 0) && (firstFactor != i)) {
            break;
        } else {
            firstFactor = i;
        }
    }
    if (firstFactor === (x - 1)) {
        alert(`The number ${x} is prime.`);
    } else {
        firstFactor ++;
        alert(`The number ${x} is not prime. It's first factor is ${i}.`);
    }
};
isPrime(x);